typescript/api/controllers/TranslationController.ts

```diff
@@ -0,0 +1,202 @@
+            const { _meta, ...rest } = payload;
+            return res.json(rest);
+          } catch (_e) {
+            // fallback
+          }
+        }
+        return res.json(payload);
+      } catch (err) {
+        sails.log.error('Error in TranslationController.getNamespace:', err);
+        return res.serverError(err);
+      }
+    }
+
+    /**
+     * Return the list of supported languages for the current branding/portal.
+     * Combines configured languages with any detected from DB bundles.
+     * Returns a list of objects with code, displayName, and enabled.
+     */
+    public async getLanguages(req, res) {
+      try {
+        const brandingName = req.params.branding;
+        const branding = BrandingService.getBrand(brandingName);
+        if (!branding) {
+          return res.badRequest({ message: `Unknown branding: ${brandingName}` });
+        }
+
+        const langCodes = new Set<string>();
+        const configured = sails?.config?.i18n?.next?.init?.supportedLngs;
+        if (Array.isArray(configured)) configured.forEach((l: string) => l && langCodes.add(l));
+
+        // From DB bundles using I18nEntriesService
+        const bundles: any[] = await I18nEntriesService.listBundles(branding);
+        bundles.forEach((b: any) => b?.locale && langCodes.add(b.locale));
+
+        const codes = Array.from(langCodes);
+        const bundleMap = new Map(bundles.map(b => [b.locale, b]));
+
+        const list = await Promise.all(codes.map(async code => {
+          const bundle = bundleMap.get(code);
+          return {
+            code: code,
+            displayName: bundle?.displayName || await I18nEntriesService.getLanguageDisplayName(code),
+            enabled: bundle?.enabled !== false // defaults to true if not set
+          };
+        }));
+
+        list.sort((a, b) => a.displayName.localeCompare(b.displayName));
+        return res.json(list);
+      } catch (err) {
+        sails.log.error('Error in TranslationController.getLanguages:', err);
+        return res.serverError(err);
+      }
+    }
+
+    /**
+     * Angular app endpoints (CSRF-enabled by default)
+     */
+    public async listEntriesApp(req, res) {
+      try {
+        const brandingName = req.params.branding;
+        const branding = BrandingService.getBrand(brandingName);
+        if (!branding) return res.badRequest({ message: `Unknown branding: ${brandingName}` });
+        const locale = req.param('locale');
+        const namespace = req.param('namespace') || 'translation';
+        const keyPrefix = req.param('keyPrefix');
+        const entries = await I18nEntriesService.listEntries(branding, locale, namespace, keyPrefix);
+        return res.json(entries);
+      } catch (err) {
+        sails.log.error('Error in TranslationController.listEntriesApp:', err);
+        return res.serverError(err);
+      }
+    }
+
+    public async setEntryApp(req, res) {
+      try {
+        const brandingName = req.params.branding;
```

Comment:
I don't know if it matters, but some request parameters use dot-accessors, and some use a method call. Should they be consistent?